usvm-ts/src/main/kotlin/org/usvm/machine/expr/CallApproximations.kt

```diff
@@ -0,0 +1,879 @@
+        val originalLength = memory.read(originalLengthLValue)
+
+        // Copy the original array to the result array
+        memory.memcpy(
+            srcRef = array,
+            dstRef = resultArray,
+            type = arrayType,
+            elementSort = elementSort,
+            fromSrc = 0.toBv().asExpr(sizeSort),
+            fromDst = 0.toBv().asExpr(sizeSort),
+            length = originalLength,
+        )
+
+        // Handle each argument in the `concat` call
+        var totalLength = originalLength
+        for (arg in args) {
+            // For array arguments, copy their elements to the result array
+            if (arg.sort == addressSort) {
+                // TODO: handle empty type stream
+                val argType = memory.typeStreamOf(arg.asExpr(addressSort)).first()
+                if (argType is EtsArrayType) {
+                    val argLengthLValue = mkArrayLengthLValue(arg.asExpr(addressSort), argType)
+                    val argLength = memory.read(argLengthLValue)
+
+                    // Copy the elements of the argument array to the result array
+                    memory.memcpy(
+                        srcRef = arg.asExpr(addressSort),
+                        dstRef = resultArray,
+                        type = arrayType,
+                        elementSort = elementSort,
+                        fromSrc = 0.toBv().asExpr(sizeSort),
+                        fromDst = totalLength,
+                        length = argLength,
+                    )
+
+                    // Add the length of the argument array to the total length
+                    totalLength = mkBvAddExpr(totalLength, argLength)
+
+                    continue
+                }
+            }
+
+            // For non-array arguments, treat them as a single element
+            val newIndexLValue = mkArrayIndexLValue(
+                sort = elementSort,
+                ref = resultArray,
+                index = totalLength,
+                type = arrayType,
+            )
+            memory.write(newIndexLValue, arg.asExpr(elementSort), guard = trueExpr)
+            totalLength = mkBvAddExpr(totalLength, 1.toBv())
+        }
+
+        // Set the length of the result array
+        val resultLengthLValue = mkArrayLengthLValue(resultArray, arrayType)
+        memory.write(resultLengthLValue, totalLength, guard = trueExpr)
+
+        // Return the new concatenated array
+        resultArray
+    }
+}
+
+/**
+ * Handles the `Array.indexOf(searchElement)` method call.
+ * Returns the index of the first occurrence of `searchElement` in the array, or `-1` if not found.
+ *
+ * https://tc39.es/ecma262/multipage/indexed-collections.html#sec-array.prototype.indexof
+ */
+private fun TsExprResolver.handleArrayIndexOf(
+    expr: EtsInstanceCallExpr,
+    arrayType: EtsArrayType,
+    elementSort: USort,
+): UExpr<*>? = with(ctx) {
+    val array = resolve(expr.instance)?.asExpr(addressSort) ?: return null
```

Comment:
Add a log message in every place we overapproximate here that we called a function that is not fully implemented in USVM